bindings/java/src/main/native/jucx_common_def.cc

```diff
@@ -136,6 +136,17 @@ void jucx_request_init(void *request)
      struct jucx_context *ctx = (struct jucx_context *)request;
      ctx->callback = NULL;
      ctx->jucx_request = NULL;
+     ctx->is_completed = false;
+     ucs_spinlock_init(&ctx->lock);
+}
+
+static void jucx_request_reset(void *request)
+{
+    struct jucx_context *ctx = (struct jucx_context *)request;
+    ctx->callback = NULL;
+    ctx->jucx_request = NULL;
+    ctx->is_completed = false;
+    ucp_request_free(request);
```

Comment:
still code duplication
need to remove ucp_request_free() from jucx_request_reset(). Then, reuse jucx_request_reset() from jucx_request_init()